invisible_cities/reco/deconv_functions_test.py

```diff
@@ -0,0 +1,179 @@
+import os
+import string
+import random
+import numpy  as np
+import pandas as pd
+
+
+from hypothesis              import given
+from hypothesis              import reproduce_failure
+from hypothesis.strategies   import text
+from hypothesis.strategies   import floats
+from hypothesis.strategies   import tuples
+from hypothesis.strategies   import composite
+from hypothesis.strategies   import lists
+from hypothesis.extra.numpy  import arrays
+
+from .. reco.deconv_functions import cut_and_redistribute_df
+from .. reco.deconv_functions import drop_isolated_sensors
+from .. reco.deconv_functions import interpolate_signal
+from .. reco.deconv_functions import deconvolution_input
+from .. reco.deconv_functions import deconvolve
+from .. reco.deconv_functions import richardson_lucy
+from .. reco.deconv_functions import InterpolationMethod
+
+from .. core.core_functions   import in_range
+from .. core.core_functions   import shift_to_bin_centers
+from .. core.testing_utils    import assert_dataframes_close
+
+from ..   io.dst_io           import load_dst
+
+from scipy.stats             import multivariate_normal
+
+characters = string.ascii_letters
+
+@composite
+def dst(draw, dimension=[3,10]):
+    columns = draw(lists(text(characters, min_size=5), min_size=dimension[0], max_size=dimension[1], unique=True))
+    values  = draw(lists(arrays(float, 100, floats(1, 1e3, allow_nan=False, allow_infinity=False)), min_size=len(columns), max_size=len(columns)))
+
+    return pd.DataFrame({k:v for k, v in zip(columns, values)})
+
+
+@given(dst())
+def test_cut_and_redistribute_df(df):
+    cut_var       = random.choice (df.columns)
+    redist_var    = random.choices(df.columns, k=3)
+    cut_val       = round(df[cut_var].mean(), 3)
+    cut_condition = f'{cut_var} > {cut_val:.3f}'
+    cut_function  = cut_and_redistribute_df(cut_condition, redist_var)
+    df_cut        = cut_function(df)
+    df_cut_manual = df.loc[df[cut_var].values > cut_val, :]
+    for var in redist_var:
+        df_cut_manual.loc[:, var] = df_cut_manual.loc[:, var] * df.loc[:, var].sum() /  df_cut_manual.loc[:, var].sum()
```

Comment:
I think you can remove the loop if you just replace `var` with `redist_var` in the second line.